### PR TITLE
AC-992: Improved Dark Mode

### PR DESCRIPTION
**Pull request by prateek27das**
AC-992 : Improved Dark Mode

## Description of what I changed
 In Dark mode, theme of custom dialog is light which is straining to eyes in dark.
 I Changed it to dark theme.

## Issue I worked on
JIRA Issue: https://issues.openmrs.org/browse/AC-992

## Short Video demonstrating my work
https://drive.google.com/file/d/183Wy5PNdKAqUI5Titx3w2LAtunwWMfNf/view?usp=sharing

## Screenshot after the improvement 
Logging out Select Location

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by prateek27das** [comment]
hi @LuGO0 can you please review now ?

**Comment by LuGO0** [comment]
Sure eill review !

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java [hunk 1]
```diff
@@ -180,9 +180,9 @@ public final void setOnBackListener() {
                     dismiss();
                 } else {
                     ((LoginFragment) getActivity()
-                        .getSupportFragmentManager()
-                        .findFragmentById(R.id.loginContentFrame))
-                        .hideURLDialog();
+                            .getSupportFragmentManager()
+                            .findFragmentById(R.id.loginContentFrame))
+                            .hideURLDialog();
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java:185]: remove these formatting changes. only format portions of the code you changed.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java [hunk 2]
```diff
@@ -194,7 +194,7 @@ public final void setBorderless() {
         getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
 
         int marginWidth = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, TYPED_DIMENSION_VALUE,
-            OpenMRS.getInstance().getResources().getDisplayMetrics());
+                OpenMRS.getInstance().getResources().getDisplayMetrics());
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java:197]: ditto

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java [hunk 3]
```diff
@@ -209,7 +209,7 @@ public final void setProgressDialogWidth() {
...
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java:212]: ditto

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java [hunk 4]
```diff
@@ -380,16 +380,16 @@ private boolean isDialogAvailable() {
...
                     dismiss();
                     break;
                 case LOGIN:
                     ((LoginFragment) getActivity()
-                        .getSupportFragmentManager()
-                        .findFragmentById(R.id.loginContentFrame))
-                        .login(true);
+                            .getSupportFragmentManager()
+                            .findFragmentById(R.id.loginContentFrame))
+                            .login(true);
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java:392]: ditto

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java [hunk 5]
```diff
@@ -380,16 +380,16 @@ private boolean isDialogAvailable() {
             switch (action) {
                 case DISMISS_URL_DIALOG:
                     ((LoginFragment) getActivity()
-                        .getSupportFragmentManager()
-                        .findFragmentById(R.id.loginContentFrame))
-                        .hideURLDialog();
+                            .getSupportFragmentManager()
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java:383]: ditto

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java [hunk 6]
```diff
@@ -271,7 +271,7 @@ private void addSingleChoiceItemsListView(List<String> locationList) {
         LinearLayout field = (LinearLayout) mInflater.inflate(R.layout.openmrs_single_choice_list_view, null);
         locationListView = field.findViewById(R.id.singleChoiceListView);
         locationListView.setAdapter(new ArrayAdapter<>(getActivity(),
-            R.layout.row_single_checked_layout, locationList));
+                R.layout.row_single_checked_layout, locationList));
```
Comment (LuGO0) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CustomFragmentDialog.java:274]: ditto

File: openmrs-client/src/main/res/layout/openmrs_text_view_field.xml [hunk 7]
```diff
@@ -14,19 +14,24 @@
...
-        />
+    android:orientation="vertical"
+    android:background="@color/black_dark_mode"
+    android:layout_width="fill_parent"
+    android:layout_height="fill_parent">
+    <RelativeLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        style="@style/BackgroundTheme"
+        >
```
Comment (LuGO0) [inline openmrs-client/src/main/res/layout/openmrs_text_view_field.xml:25]: wrong formatting this should be in above line?

File: openmrs-client/src/main/res/layout/openmrs_text_view_field.xml [hunk 8]
```diff
@@ -14,19 +14,24 @@
...
+        <TextView
+            android:id="@+id/openmrsTextView"
+            android:layout_width="fill_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="20dp"
+            android:layout_marginLeft="15dp"
+            android:layout_marginRight="15dp"
+            android:textSize="22sp"
+            />
```
Comment (LuGO0) [inline openmrs-client/src/main/res/layout/openmrs_text_view_field.xml:34]: similar issue.

**Review by LuGO0 — CHANGES_REQUESTED** [review]
try getting the openmrsFormatter before making any formatting changes it would be easy for you to maintain the correct formatting.
other things LGTM !

**Comment by prateek27das** [comment]
@LuGO0 I tried to import the OpenMRS Formatter using Eclipse Code Formatter. I followed the "Contribution GuideLines". But the formatter is not coming to the Scheme Portion. 

I am facing the same issue as it was discussed before in OpenMRS talk 
here is the link of that talk - https://talk.openmrs.org/t/how-to-import-openmrsformatter-xml-in-android-studio/31549

Please help me to set the OpenMRS Formatter !!!

This is my Eclipse Code Formatter settings


Screenshot 2022-03-02 at 5 22 41 PM

**Comment by LuGO0** [comment]
I think you just need to install the eclipse code formatter plugin which helps in this, not really sure about the whole process since haven't done it in a while, just try it with the plugin if you are not able to do it, I will try to lay down some elaborate steps for the same.

**Comment by prateek27das** [comment]
@LuGO0 ,I have installed the Eclipse Code formatter plugin and do the required as described in the "Contribution GuideLines"

**Comment by LuGO0** [comment]
Screenshot 2022-03-03 000809
I did it using this settings button and imported the style using the openmrsformatter
 XML file

**Comment by LuGO0** [comment]
try this out https://github.com/krasa/EclipseCodeFormatter#instructions

**Comment by LuGO0** [comment]
For this PR you can just revert those changes manually so that I can merge them.

**Comment by prateek27das** [comment]
I am setting up the formatter  
As you said earlier ,I reverted those changes manually. for the  CustomFragmentDialog.java as of now. 

can you guide me formatting the "openmrs_text_view_field.xml" manually.

**Pull request merged**